lib/tx.js

```diff
@@ -0,0 +1,281 @@
+      link.hints.allow.length === 1) {
+    var method = link.hints.allow[0];
+    switch (method) {
+
+      case 'GET':
+        return function() {
+          return http.get(sdk, link.href);
+        };
+
+      case 'POST':
+        return function(opts) {
+          if (ref && ref.isPolling) {
+            ref.isPolling = false;
+          }
+
+          var data = addStateToken(res, opts);
+
+          if (res.status === 'MFA_ENROLL') {
+            // Add factorType and provider
+            util.extend(data, {
+              factorType: obj.factorType,
+              provider: obj.provider
+            });
+          }
+
+          var href = link.href;
+          if (data.rememberDevice !== undefined) {
+            if (data.rememberDevice) {
+              href += '?rememberDevice=true';
+            }
+            data = util.omit(data, 'rememberDevice');
+
+          } else if (data.profile &&
+                    data.profile.updatePhone !== undefined) {
+            if (data.profile.updatePhone) {
+              href += '?updatePhone=true';
+            }
+            data.profile = util.omit(data.profile, 'updatePhone');
+          }
+
+          return postToTransaction(sdk, href, data);
+        };
+    }
+  }
+}
+
+function links2fns(sdk, res, obj, ref) {
+  var fns = {};
+  for (var linkName in obj._links) {
+    if (!obj._links.hasOwnProperty(linkName)) {
+      continue;
+    }
+
+    var link = obj._links[linkName];
+    
+    if (linkName === 'next') {
+      linkName = link.name;
+    }
+
+    if (link.type) {
+      fns[linkName] = link;
+      continue;
+    }
+
+    switch (linkName) {
+      // poll is only found at the transaction
+      // level, so we don't need to pass the link
+      case 'poll':
+        fns.poll = getPollFn(sdk, res, ref);
+        break;
+
+      default:
+        var fn = link2fn(sdk, res, obj, link, ref);
+        if (fn) {
+          fns[linkName] = fn;
+        }
+    }
+  }
+  return fns;
+}
+
+function flattenEmbedded(sdk, res, obj, ref) {
+  obj = obj || res;
+  obj = util.clone(obj);
+
+  if (Array.isArray(obj)) {
+    var objArr = [];
+    for (var o = 0, ol = obj.length; o < ol; o++) {
+      objArr.push(flattenEmbedded(sdk, res, obj[o], ref));
+    }
+    return objArr;
+  }
+
+  var embedded = obj._embedded || {};
+
+  for (var key in embedded) {
+    if (!embedded.hasOwnProperty(key)) {
+      continue;
+    }
+
+    // Flatten any nested _embedded objects
+    if (util.isObject(embedded[key]) || Array.isArray(embedded[key])) {
+      embedded[key] = flattenEmbedded(sdk, res, embedded[key], ref);
+    }
+  }
+
+  // Convert any links on the embedded object
+  var fns = links2fns(sdk, res, obj, ref);
+  util.extend(embedded, fns);
+
+  obj = util.omit(obj, '_embedded', '_links');
+  util.extend(obj, embedded);
+  return obj;
+}
+
+function AuthTransaction(sdk, res) {
+  if (res) {
+    this.data = res;
+    util.extend(this, flattenEmbedded(sdk, res, res, {}));
```

Comment:
I'd love to remove AuthTransaction into its own file, but this dependency on flattenEmbedded means `AuthTransaction.js` would depend on `tx.js` and `postToTransaction` depends on AuthTransaction, so `tx.js` would depend on `AuthTransaction.js`, creating a circular dependency. [ES6 supports certain cyclic dependencies](http://exploringjs.com/es6/ch_modules.html#_cyclic-dependencies-in-ecmascript-6), so we can pull out AuthTransaction when we move to ES6.